deeplake/core/vectorstore/deep_memory.py

```diff
@@ -0,0 +1,348 @@
+        corpus_path = self.dataset.path
+        queries_path = corpus_path + "_queries"
+
+        if embedding_function is None and self.embedding_function is None:
+            raise ValueError(
+                "Embedding function should be specifed either during initialization or during training."
+            )
+
+        if embedding_function is None and self.embedding_function is not None:
+            embedding_function = self.embedding_function.embed_documents
+
+        queries_vs = VectorStore(
+            path=queries_path,
+            overwrite=True,
+            runtime={"tensor_db": True},
+            embedding_function=embedding_function,
+            token=token,
+        )
+
+        queries_vs.add(
+            text=[query for query in queries],
+            metadata=[
+                {"relevance": [(doc_id, 1) for doc_id in relevance]}
+                for relevance in relevance
+            ],
+            embedding_data=[query for query in queries],
+        )
+
+        # do some rest_api calls to train the model
+        response = self.client.start_taining(
+            corpus_path=corpus_path,
+            queries_path=queries_path,
+        )
+        return response["job_id"]
+
+    def cancel(self, job_id: str):
+        """Cancel a training job on DeepMemory managed service.
+
+        Args:
+            job_id (str): job_id of the training job.
+
+        Returns:
+            bool: True if job was cancelled successfully, False otherwise.
+        """
+        try:
+            self.client.cancel_job(job_id=job_id)
+            print(f"Job with job_id='{job_id}' was sucessfully cancelled!")
+            return True
+        except Exception as e:
+            print(f"Job with job_id='{job_id}' was not cancelled! Error: {e}")
+            return False
+
+    def delete(self, job_id: str):
+        """Delete a training job on DeepMemory managed service.
+
+        Args:
+            job_id (str): job_id of the training job.
+
+        Returns:
+            bool: True if job was deleted successfully, False otherwise.
+        """
+        try:
+            self.client.delete_job(job_id=job_id)
+            print(f"Job with job_id='{job_id}' was sucessfully deleted!")
+            return True
+        except Exception as e:
+            print(f"Job with job_id='{job_id}' was not deleted! Error: {e}")
+            return False
+
+    def status(self, job_id: str):
+        """Get the status of a training job on DeepMemory managed service.
+
+        Args:
+            job_id (str): job_id of the training job.
+        """
+        self.client.check_status(job_id=job_id)
+
+    def list_jobs(self):
+        """List all training jobs on DeepMemory managed service."""
+        response = self.client.list_jobs(dataset_path=self.dataset.path)
+        return response
+
+    def evaluate(
+        self,
+        relevance: List[List[str]],
```

Comment:
again `relevance: List[List[Tuple[str, int]]]` contains an integer, this integer is going to help us later down the road, but now we need to get the API used to that.

Reply:
Sorry I'm not the best to give an opinion here. At the demo, I was suggesting supporting multiple options, ranging from the simples one to the more complex+future proof one. Since we're likely out of time, just pick the one you need right now.